CONTRIBUTING.md

```diff
@@ -108,6 +108,163 @@ To learn about the viewer, run:
 cargo run -p rerun -- --help
 ```
 
+## Tests
+
+There are various kinds of automated tests throughout the repository.
+If not noted otherwise, all tests run automated on CI, however their frequency (per PR, on `main`, nightly, etc.) and platform coverage may vary.
+
+### Rust tests
+
+```sh
+cargo nextest run --all-targets --all-features
+cargo test --all-features --doc
+```
+or alternatively:
+```sh
+cargo test --all-targets --all-features
+```
+
+Runs both unit & integration tests for Rust crates, including the Rerun viewer.
+
+Tests are written using the standard `#[test]` attribute.
+
+#### `insta` snapshot tests
+
+Some of the tests in the `rerun` family of crates are [`insta`](https://docs.rs/insta/latest/insta/) snapshot tests.
+These tests work by comparing a textual output of a test against a checked-in reference.
+
+They run as part of the regular Rust test suite, no extra action is required to include them in a test run.
+
+#### Image comparision tests
+
+Some of the tests in the `rerun` family of crates are image comparision tests.
+These tests work by rendering an image an image and then comparing it with a checked-in reference image.
+
+They run as part of the regular Rust test suite, no extra action is required to include them in a test run.
+
+Comparision tests are driven by [egui_kittest](https://github.com/rerun-io/egui_kittest)'s `Harness::snapshot` method.
+Typically, we use [TestContext](rerun/crates/viewer/re_viewer_context/src/test_context.rs) in order to mock
+relevant parts of the Rerun viewer.
+
+##### Comparing results & updating images
+
+Each run of the comparision tests will produce new images that are saved to the comparision images.
+(typically at `<your-test.rs>/snapshots`)
+
+Upon failure, additionally `diff.png` file is added that highlights all differences between the reference and the new image.
+In order to update reference with the new image, run with `UPDATE_SNAPSHOTS=1` environment variable set.
+
+Use `pixi run snapshots` to compare the results of all failed tests in Rerun.
+
+##### Guidelines for writing image comparision tests
+
+* avoid! Whenever **possible** prefer regular Rust tests or `insta` snapshot tests over image comparision tests.
+* images should…
+  * …be checked in as LFS file
+  * …depict exactly what's tested and nothing else
+  * …have a low resolution to avoid growth in repo size
+  * …have a low comparision threshold to avoid false positives
```

Comment:
low comparision threshold -> False negatives?

Reply:
false positive: the test is incorrectly marked as passing

... better writing that out, too confusing